### PR TITLE
lesson 1.3

### PR DESCRIPTION
**Pull request by kmg37**
refactored todolist.js, added event handlers and useState for form

### CODE REVIEW

File: src/TodoList.js [hunk 1]
```diff
@@ -9,8 +10,8 @@ const todoList = [
 function TodoList() {
   return (
     <ul>
-      {todoList.map((item) => (
-        <li key={item.id}>{item.title}</li>
+      {todoList.map((props) => (
+        <TodoListItem id={props.id} title={props.title} />
```
Comment (mattcarlotta) [inline src/TodoList.js]: Requires changes:
When iterating over an array of items, React needs a `key` property to distinguish which item in the list belongs to which item in the DOM: https://react.dev/learn/rendering-lists

As mentioned in the link above, not providing a key with throw an error in the console:
> Warning: Each child in a list should have a unique “key” prop.

File: src/TodoListItem.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+function TodoListItem(props) {
```
Comment (mattcarlotta) [inline src/TodoListItem.js:3]: Answering question about props naming:
In this case, `props` can be named anything, but it’s mostly a convention that was established before React moved to hooks. Before hooks, React used classes, where `this.props` was used to distinguish between values that were supplied as arguments to a React class component versus `this.state` stateful values created within a React class component. This convention was continued when React moved to hooks to keep it inline with classes. That said, unlike classes, there’s really nothing in hooks that is preventing you from naming it to something else.

**Review by mattcarlotta — CHANGES_REQUESTED** [review]
One minor change is required as mentioned below. Otherwise, everything else is looking good.

**Comment by kmg37** [comment]
@mattcarlotta thanks for the information and reference! I've added the key property, looks good in the console now

**Review by mattcarlotta — APPROVED** [review]
Looks good to me!

**Pull request merged**